facturx/tests/test_facturx.py

```diff
@@ -1,6 +1,60 @@
 import os
 import unittest
-from facturx.facturx import FacturX
+from facturx.facturx import *
+import xml.etree.ElementTree as ET
+
+
+class XmlTree:
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def convert_string_to_tree( xmlString):
+
+        return ET.fromstring(xmlString)
+
+    def xml_compare(self, x1, x2):
+        if x1.tag != x2.tag:
+            # print('Tags do not match: %s and %s' % (x1.tag, x2.tag))
+            return False
+        for name, value in x1.attrib.items():
+            if x2.attrib.get(name) != value:
+                # print('Attributes do not match: %s=%r, %s=%r'
+                #              % (name, value, name, x2.attrib.get(name)))
+                return False
+        for name in x2.attrib.keys():
+            if name not in x1.attrib:
+                # print('x2 has an attribute x1 is missing: %s'
+                #              % name)
+                return False
+        if not self.text_compare(x1.text, x2.text):
+            # print('text: %r != %r' % (x1.text, x2.text))
+            return False
+        if not self.text_compare(x1.tail, x2.tail):
+            # print('tail: %r != %r' % (x1.tail, x2.tail))
+            return False
+        cl1 = x1.getchildren()
+        cl2 = x2.getchildren()
+        if len(cl1) != len(cl2):
+            # print('children length differs, %i != %i'
+            #              % (len(cl1), len(cl2)))
+            return False
+        i = 0
+        for c1, c2 in zip(cl1, cl2):
+            i += 1
+            if not self.xml_compare(c1, c2):
+                # print('children %i do not match: %s'
+                #              % (i, c1.tag))
+                return False
+        return True
+
+    def text_compare(self, t1, t2):
+        if not t1 and not t2:
+            return True
+        if t1 == '*' or t2 == '*':
+            return True
+        return (t1 or '').strip() == (t2 or '').strip()
+
 
 class TestReading(unittest.TestCase):
     def setUp(self):
```

Comment:
I believe `setUp()` is a reserved method that is automatically run BEFORE each test. You can't call it somewhere in the middle.

Reply:
Yes, `setUp()` automatically runs before each test. I haven't faced any problem in calling it in between a function. But maybe it is not a good practice.
I can remove the `setUp()` calling, should I?

Reply:
I can't see the purpose of running the code in `setUp()` before the test and I can't see the purpose in between. What does it do?

From the [docs](https://docs.python.org/2/library/unittest.html):

> The setUp() and tearDown() methods allow you to define instructions that will be executed before and after each test method.

Reply:
In our case 
 - At the beginning: it is discovery the directory where sample invoices are kept. 
 - In between: it is required because I am creating a new PDF and then checking whether xml has been embedded or not. Hence, it needs to rediscover the files (because of the new file created).

Reply:
Github seems to show old code. I see `self.test_files = os.listdir(self.test_files_dir)` locally, but something else here on Github. Assigning something to `self` is ok. Calling it in-between is confusing and shouldn't be done.

Reply:
You are only using `self.test_files` in 2 methods. The setUp function is called in all other methods as well without being used. Seems like a big waste and faulty layout.